htdocs/product/card.php

```diff
@@ -2282,6 +2282,10 @@
 					print '<a class="butAction" href="'.$_SERVER["PHP_SELF"].'?action=clone&amp;id='.$object->id.'">'.$langs->trans("ToClone").'</a>';
 				}
 			}
+
+			if ( ! empty($conf->productbatch->enabled) && $object->hasBatch() ) {
```

Comment:
This feature is related to stock. And this link/button already exists on tab "Stock".

Reply:
it was a demand from users to have a "shortcut" on the product tab to avoid multiple clicks.
And in the logic of the users, equipments are more linked to the product than to stock.
@lmag

Reply:
@altatof maybe add to dolitools ;)

Reply:
Yes shortcuts is a good idea, but everybody wants his own personal shortcut. A company without stock mgmt but with multi-language will want a shortcut to add a product translation. Another who only deals with purchase will want a shortcut to add a supplier price...
So a global shortcut system giving access to actions directly from the tabs for example would be nice ?
And in your code you didn't check the user permission on stock.

Reply:
hello @atm-maxime 
_So a global shortcut system giving access to actions directly from the tabs for example would be nice ?_
Very good idea ! How do you imagine that ?

Reply:
I agree with @maxime.
There is a ton of user request to have shortcut depending of what they used the most. But we can't add all combinations of desired shortcuts in application everywhere. Position of feature must remains at the best logical place (and to do a stock movement, the best place is the "Stock" tab, not the general information tab.
Currently, feature to add your own shortcut is an external module that add link by using the hook.

Reply:
> How do you imagine that ?

Currently we have static tabs everywhere and action buttons at the bottom of each page. If we add on every tab a link (or on right click?) showing available actions this would be helpful I think.

Reply:
I am not sure I understood your proposals exactly.
The idea of ​​being able to customize all the tabs seems practical to me. 
I will review with @altatof the demand for stocks because it seems to me that @eldy is right (stocks in stock) ! In case we will upload screens to be sure we are talking of the same thing.